packages/studio-ui/src/web/views/FlowBuilder/common/TransitionItem.tsx

```diff
@@ -65,12 +69,14 @@ export default class TransitionItem extends Component<Props> {
 
   render() {
     let raw
-    const { position } = this.props
-    const { condition, caption } = this.props.transition
+    const { position, transition, currentLang, defaultLang } = this.props
+    const { condition } = transition
 
-    if (caption) {
+    const caption = transition[`caption$${currentLang}`] || transition[`caption$${defaultLang}`] || transition.caption
+
+    if (caption && typeof caption === 'string') {
       const vars = {}
-      const htmlTpl = caption.replace(/\[(.+)]/gi, (x) => {
+      const htmlTpl = lang.tr(caption).replace(/\[(.+)]/gi, (x) => {
```

Comment:
`lang.tr` looks up in the i18n translation files. are you sure you want to do this ?

Reply:
There are cases where we want to use translation files; for instance, for the onFailure string, it will not be dynamic, and we can use translation files instead to prevent node properties pollution